lib/rules/no-use-before-define.js

```diff
@@ -58,10 +70,30 @@ module.exports = function(context) {
         });
     }
 
+
+    /**
+     * Validates variables inside of a node's scope.
+     * @param {ASTNode} node The node to check.
+     * @returns {void}
+     * @private
+     */
+    function findVariablesInNode() {
+        var scope = context.getScope();
+        findVariables(scope);
+    }
+
     return {
-        "Program": findVariables,
-        "FunctionExpression": findVariables,
-        "FunctionDeclaration": findVariables,
-        "ArrowFunctionExpression": findVariables
+        "Program": function() {
+            var scope = context.getScope();
+            findVariables(scope);
+
+            // both Node.js and Modules have an extra scope
+            if (context.ecmaFeatures.globalReturn || context.ecmaFeatures.modules) {
+                findVariables(scope.childScopes[0]);
+            }
+        },
+        "FunctionExpression": findVariablesInNode,
```

Comment:
Can't actually do `scope = scope || context.getScope()` since the AST node is passed as an argument.